File: docs/snippets/all/tutorials/annotation_context.py
import dalaran as dl

dl.init("dalaran_example_annotation_context_connections")

# Annotation context with two classes, using two labeled classes, of which
# ones defines a color.
dl.log(
    "masks",  # Applies to all entities below "masks".
    dl.AnnotationContext(
        [
            dl.AnnotationInfo(id=0, label="Background"),
            dl.AnnotationInfo(id=1, label="Person", color=(255, 0, 0)),
        ],
    ),
    static=True,
)

# Annotation context with simple keypoints & keypoint connections.
dl.log(
    "detections",  # Applies to all entities below "detections".
    dl.ClassDescription(
        info=dl.AnnotationInfo(0, label="Snake"),
        keypoint_annotations=[
            dl.AnnotationInfo(id=i, color=(0, 28 * i, 0)) for i in range(10)
        ],
        keypoint_connections=[(i, i + 1) for i in range(9)],
    ),
    static=True,
)

File: docs/snippets/all/tutorials/annotation_context.cpp
#include <dalaran.hpp>

int main(int argc, char* argv[]) {
    const auto rec =
        dalaran::RecordingStream("dalaran_example_annotation_context_connections");
    rec.spawn().exit_on_failure();

    // Annotation context with two classes, using two labeled classes, of which ones defines a
    // color.
    rec.log_static(
        "masks",
        dalaran::AnnotationContext({
            dalaran::AnnotationInfo(0, "Background"),
            dalaran::AnnotationInfo(1, "Person", dalaran::Rgba32(255, 0, 0)),
        })
    );

    // Annotation context with simple keypoints & keypoint connections.
    std::vector<dalaran::AnnotationInfo> keypoint_annotations;
    for (uint16_t i = 0; i < 10; ++i) {
        keypoint_annotations.push_back(dalaran::AnnotationInfo(
            i,
            dalaran::Rgba32(0, static_cast<uint8_t>(28 * i), 0)
        ));
    }

    std::vector<dalaran::KeypointPair> keypoint_connections;
    for (uint16_t i = 0; i < 9; ++i) {
        keypoint_connections.push_back(dalaran::KeypointPair(i, i + 1));
    }

    rec.log_static(
        "detections", // Applies to all entities below "detections".
        dalaran::AnnotationContext({dalaran::ClassDescription(
            dalaran::AnnotationInfo(0, "Snake"),
            keypoint_annotations,
            keypoint_connections
        )})
    );
}
